src/io/inputs.jl

```diff
@@ -57,54 +57,50 @@ function load_scenarios(domain::Domain, filepath::String)::DataFrame
-        loaded = NamedDimsArray(data[attr]; Source=site_order, Sink=site_order)
-    catch err
-        if isa(err, KeyError)
-            @warn "Provided file $(data_fn) did not have reef_siteid! There may be a mismatch in sites."
-            if size(loaded, 2) != nrow(site_data)
-                @warn "Mismatch in number of sites ($(data_fn)).\nTruncating so that data size matches!"
-
-                # Subset down to number of sites
-                tmp = selectdim(data[attr], 2, 1:nrow(site_data))
-                loaded = NamedDimsArray(tmp; Source=site_order, Sink=site_order)
-            end
-        else
-            rethrow(err)
-        end
-    end
-
-    return loaded
-end
-
 """
-    load_nc_data(data_fn::String, attr::String, site_data::DataFrame)::NamedDimsArray
+    load_nc_data(data_fn::String, attr::String, site_data::DataFrame; dim_names::Vector{Symbol}=[], dim_names_replace::Vector=[])::YAXArray
 
-Load cluster-level data for a given attribute in a netCDF as a NamedDimsArray.
+Load cluster-level data for a given attribute in a netCDF.
 """
-function load_nc_data(data_fn::String, attr::String, site_data::DataFrame)::NamedDimsArray
-    local loaded_data::NamedDimsArray
-
+function load_nc_data(
+    data_fn::String,
+    attr::String,
+    site_data::DataFrame;
+    dim_names::Vector=[],
+    dim_names_replace::Vector=[],
+)::YAXArray
     NetCDF.open(data_fn; mode=NC_NOWRITE) do nc_file
         data::Array{<:AbstractFloat} = NetCDF.readvar(nc_file, attr)
-        dim_names::Vector{Symbol} = Symbol[
-            Symbol(dim.name) for dim in nc_file.vars[attr].dim
-        ]
-        dim_labels::Vector{Union{UnitRange{Int64},Vector{String}}} = _nc_dim_labels(
-            data_fn, data, nc_file
-        )
+
+        dim_labels = _nc_dim_labels(data_fn, data, nc_file)
+
+        # Handle dim_names
+        if isempty(dim_names)
+            dim_names = [Symbol(dim.name) for dim in nc_file.vars[attr].dim]
+        end
+
+        if !isempty(dim_names_replace)
+            replace!(dim_names, dim_names_replace...)
+        end
+
+        # Order sites
+        if :sites ∈ dim_names
+            sites_dim::Int64 = findfirst(x -> x == :sites, dim_names)
+            sites::Vector{String} = dim_labels[sites_dim]
+            sites_sort_idx = sortperm(sites)
+
+            dim_labels[sites_dim] = dim_labels[sites_dim][sites_sort_idx]
+
+            # Guarantees that even if sites dimension changes this sorting still works
+            selector::Vector{Any} = fill(:, length(dim_labels))
+            selector[sites_dim] = sites_sort_idx
+            data = data[selector...]
+        end
+
+        axlist = Tuple(Dim{i[1]}(i[2]) for i in zip(dim_names, dim_labels))
 
         try
-            loaded_data = NamedDimsArray(data; zip(dim_names, dim_labels)...)
+            return YAXArray(axlist, data)
         catch err
             if isa(err, KeyError)
                 n_sites = size(data, 2)
```

Comment:
The reason for this to be wrapped in a `try/catch` isn't clear to me.

Additionally, there is no return in the case of a `KeyError` either so will lead to an issue.

Reply:
I removed the try/catch because it wasn't clear to me either (it was there before and I just kept it). Let me know if you have any problems with that.

Reply:
Thank you - it was there before for a reason, but you've removed the reason for it 😉